### PR TITLE
Migrate cameras to required components

### PR DESCRIPTION
**Pull request by Jondolf**
# Objective

Yet another PR for migrating stuff to required components. This time, cameras!

## Solution

As per the [selected proposal](https://hackmd.io/tsYID4CGRiWxzsgawzxG_g#Combined-Proposal-1-Selected), deprecate `Camera2dBundle` and `Camera3dBundle` in favor of `Camera2d` and `Camera3d`.

Adding a `Camera` without `Camera2d` or `Camera3d` now logs a warning, as suggested by Cart [on Discord](https://discord.com/channels/691052431525675048/1264881140007702558/1291506402832945273). I would personally like cameras to work a bit differently and be split into a few more components, to avoid some footguns and confusing semantics, but that is more controversial, and shouldn't block this core migration.

## Testing

I ran a few 2D and 3D examples, and tried cameras with and without render graphs.

---

## Migration Guide

`Camera2dBundle` and `Camera3dBundle` have been deprecated in favor of `Camera2d` and `Camera3d`. Inserting them will now also insert the other components required by them automatically.

### CODE REVIEW

**Review by cart — APPROVED** [review]
Looks good to me!

File: examples/picking/mesh_picking.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
+//! A simple 3D scene to demonstrate mesh picking.
```
Comment (Jondolf) [inline examples/picking/mesh_picking.rs:1]: Uhhh so I did *not* mean to commit this here lmao, please hold off on merging until I remove this later today

Comment (Jondolf) [reply examples/picking/mesh_picking.rs]: Fixed

**Review by LiamGallagher737 — APPROVED** [review]

File: crates/bevy_core_pipeline/src/core_2d/camera_2d.rs [hunk 2]
```diff
@@ -17,12 +19,25 @@ use bevy_render::{
 };
 use bevy_transform::prelude::{GlobalTransform, Transform};
 
+/// A 2D camera component. Enables the 2D render graph for a [`Camera`].
 #[derive(Component, Default, Reflect, Clone, ExtractComponent)]
 #[extract_component_filter(With<Camera>)]
 #[reflect(Component, Default)]
+#[require(
```
Comment (KirmesBude) [inline crates/bevy_core_pipeline/src/core_2d/camera_2d.rs:26]: SyncToRenderWorld seems to not be added now (as far as I can tell not through any dependencies either). Is this intentional?

File: crates/bevy_core_pipeline/src/core_3d/camera_3d.rs [hunk 3]
```diff
@@ -15,12 +17,22 @@ use bevy_render::{
...
-/// Configuration for the "main 3d render graph".
-/// The camera coordinate space is right-handed x-right, y-up, z-back.
+/// A 3D camera component. Enables the main 3D render graph for a [`Camera`].
+///
+/// The camera coordinate space is right-handed X-right, Y-up, Z-back.
 /// This means "forward" is -Z.
 #[derive(Component, Reflect, Clone, ExtractComponent)]
 #[extract_component_filter(With<Camera>)]
 #[reflect(Component, Default)]
+#[require(
```
Comment (KirmesBude) [inline crates/bevy_core_pipeline/src/core_3d/camera_3d.rs:27]: SyncToRenderWorld seems to not be added now (as far as I can tell not through any dependencies either). Is this intentional?

File: crates/bevy_core_pipeline/src/core_2d/camera_2d.rs (hunk 2 above)
Comment (Jondolf) [reply crates/bevy_core_pipeline/src/core_2d/camera_2d.rs:26]: Good catch, added it for `Camera` now. This wasn't a thing when I did this migration originally, and I forgot to add it when pulling latest changes from `main` :p

**Review by KirmesBude — APPROVED** [review]

**Pull request merged**